sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/opa/testing/AbstractOpa.java

```diff
@@ -0,0 +1,345 @@
+    }
+
+    public boolean isAllow() {
+      return allow;
+    }
+
+    private boolean allow;
+
+    public String getHttpMethod() {
+      return httpMethod;
+    }
+
+    public String[] getPaths() {
+      return paths;
+    }
+
+    public boolean isMatchJWT() {
+      return matchJWT;
+    }
+
+    public String getJwt() {
+      return jwt;
+    }
+
+    public boolean isOnAnyRequest() {
+      return onAnyRequest;
+    }
+
+    private boolean onAnyRequest = false;
+    private boolean errorResponse = false;
+
+    public boolean isErrorResponse() {
+      return errorResponse;
+    }
+
+    public boolean isEmptyResponse() {
+      return emptyResponse;
+    }
+
+    public OpaResponse getAnswer() {
+      return answer;
+    }
+
+    private boolean emptyResponse = false;
+    private OpaResponse answer;
+
+    public Object getConstraint() {
+      return constraint;
+    }
+
+    private Object constraint;
+
+    static ResponseDefinitionBuilder getResponse(OpaResponse response) {
+      try {
+        return aResponse()
+            .withStatus(200)
+            .withHeader("Content-type", "application/json")
+            .withBody(OM.writeValueAsBytes(response));
+      } catch (JsonProcessingException exception) {
+        throw new IllegalStateException("Mock initialization failed");
+      }
+    }
+
+    StubBuilder onAnyRequest() {
+      onAnyRequest = true;
+      return this;
+    }
+
+    public RequestPathBuilder withHttpMethod(String httpMethod) {
+      this.httpMethod = httpMethod;
+      return this;
+    }
+
+    public RequestExtraBuilder withPath(String path) {
+      this.paths = splitPath(path);
+      return this;
+    }
+
+    public RequestExtraBuilder withJwt(String jwt) {
+      matchJWT = true;
+      this.jwt = jwt;
+      return this;
+    }
+
+    public StubBuilder allow() {
+      this.allow = true;
+      return this;
+    }
+
+    public StubBuilder deny() {
+      this.allow = false;
+      return this;
+    }
+
+    @Override
+    public BuildBuilder answer(OpaResponse answer) {
+      this.answer = answer;
+      return this;
+    }
+
+    @Override
+    public BuildBuilder emptyResponse() {
+      this.emptyResponse = true;
+      return this;
+    }
+
+    @Override
+    public BuildBuilder serverError() {
+      this.errorResponse = true;
+      return this;
+    }
+
+    @Override
+    public FinalBuilder withConstraint(Object c) {
+      this.constraint = c;
+      return this;
+    }
+
+    MappingBuilder matchAnyPostUrl() {
+      return post(urlMatching("/.*"));
+    }
+
+    MappingBuilder matchInput(String httpMethod, String[] paths) {
+      return matchAnyPostUrl()
+          .withRequestBody(matchingJsonPath("$.input.httpMethod", equalTo(httpMethod)))
+          .withRequestBody(matchingJsonPath("$.input.path", equalToJson(getJson(paths))));
+    }
+
+    static String[] splitPath(String path) {
+      if (path.startsWith("/")) {
```

Comment:
is this also working with Windows as it uses backslashes to separate paths? 
I know that this is just copy&paste from the previous code, but I'm still curious.

Reply:
as discussed- it's just the http path